pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -525,6 +526,9 @@ public boolean index(GenericRow row, @Nullable RowMetadata rowMetadata)
       }
     }
 
+    // NOTE: we msut do this before we index a single column to avoid partially indexing the row
```

Comment:
this change seems affecting non-CLP logic, can you please eloborate why this change is needed.

Reply:
It's an edge case that wasn't handled in the original Pinot code. CLP triggered this edge case in very extreme case. Specifically, if a MV column's content is larger than a chunk, Pinot will error with or without CLP.

Reply:
@Jackie-Jiang can you help review this section. is this introduce backward incompatible behavior?